src/thor/bidirectional_astar.cc

```diff
@@ -113,7 +112,8 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
                                        const GraphId& node,
                                        BDEdgeLabel& pred,
                                        const uint32_t pred_idx,
-                                       const bool from_transition) {
+                                       const bool from_transition,
+                                       const TimeInfo& time_info) {
```

Comment:
all the algorithms pass this struct around to compute time offsets at every expansion instead of having the guts of that computation in the algorithm its all in the TimeInfo struct nicely encapsulated